shell/browser/ui/views/client_frame_view_linux.cc

```diff
@@ -150,7 +150,11 @@ void ClientFrameViewLinux::Init(NativeWindowViews* window,
 }
 
 gfx::Insets ClientFrameViewLinux::GetBorderDecorationInsets() const {
-  return frame_provider_->GetFrameThicknessDip();
+  const auto insets = frame_provider_->GetFrameThicknessDip();
+  return gfx::Insets::TLBR(tiled_edges_.top ? 0 : insets.top(),
+                           tiled_edges_.left ? 0 : insets.left(),
+                           tiled_edges_.bottom ? 0 : insets.bottom(),
+                           tiled_edges_.right ? 0 : insets.right());
```

Comment:
small nit, but we can use the `tiled_edges()` accessor here. It might also help to add an explanatory comment above this logic 

```suggestion
const auto insets = frame_provider_->GetFrameThicknessDip();
  // We shouldn't draw frame decorations for the tiled edges.
  // See https://wayland.app/protocols/xdg-shell#xdg_toplevel:enum:state
  return gfx::Insets::TLBR(tiled_edges().top ? 0 : insets.top(),
                           tiled_edges().left ? 0 : insets.left(),
                           tiled_edges().bottom ? 0 : insets.bottom(),
                           tiled_edges().right ? 0 : insets.right());
```